keps/sig-instrumentation/1748-pod-resource-metrics/README.md

```diff
@@ -0,0 +1,497 @@
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [x] Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] KEP approvers have approved the KEP status as `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The current prometheus metrics exposed by the cluster have gaps that make building accurate capacity alerts, dashboards, and ad-hoc queries more difficult than necessary and complicate the average user’s understanding of the resource model. The Kubernetes resource model is fundamental to capacity planning and error triage. It should be easy to visualize and alert on core usage and capacity through a simple set of metrics. Administrators and integrators should be able to easily graph and calculate the available and consumed resources within the cluster at any time.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+Kubernetes should define a set of standard metrics that allow a user who works with the Kubernetes resource model to easily answer via instrumentation several important queries:
+
+* What is the scale of my pending scheduling queue and what resources are blocking it
+* What is the current remaining schedulable capacity of all my nodes
+* Which components are using less than they request
+* Which components on a node are the most likely to be evicted
+* Do I have workloads that are requesting significantly more initialization resources than container resources in a way that is leading to underutilization
+* Which resources on my cluster are most contended
+* Is a set of pods requesting more resources than any node has to offer
```

Comment:
I really like these questions as goals, although I think there are some higher level questions getting mixed up with lower level, more niche details. I'd group them as follows:

High level, high applicability:

- What is the current remaining schedulable capacity of all my nodes
- Which components are using less [resources] than they request
- Which components on a node are the most likely to be evicted
- Which resources on my cluster are most contended
- Is a set of pods requesting more resources than any node has to offer

Lower level, niche applicability:

- What is the scale of my pending scheduling queue and what resources are blocking it
    - Probably covered by "Which resources on my cluster are most contended?"
- Do I have workloads that are requesting significantly more initialization resources than container resources in a way that is leading to underutilization
    - Probably covered by "Which components are using less [resources] than they request?"

and possibly drop these last two questions.